awx/ui_next/src/screens/Inventory/shared/AddDropdown.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState, useRef, useEffect, Fragment } from 'react';
+import { func, string, arrayOf, shape } from 'prop-types';
+import { withI18n } from '@lingui/react';
+import { t } from '@lingui/macro';
+import {
+  Dropdown,
+  DropdownItem,
+  DropdownPosition,
+  DropdownToggle,
+} from '@patternfly/react-core';
+import { useKebabifiedMenu } from '../../../contexts/Kebabified';
+
+function AddDropdown({ dropdownItems, i18n }) {
+  const { isKebabified } = useKebabifiedMenu();
+  const [isOpen, setIsOpen] = useState(false);
+  const element = useRef(null);
+
+  useEffect(() => {
+    const toggle = e => {
+      if (!isKebabified && (!element || !element.current.contains(e.target))) {
+        setIsOpen(false);
+      }
+    };
+
+    document.addEventListener('click', toggle, false);
```

Comment:
Is there some sort of `onClick` prop or handler that we can use instead of raw dom events? I generally don't expect to see these in react components.

Reply:
update: It's for making sure the dropdown closes when you click out of it. :+1: wfm